language/vm/vm_runtime/src/block_processor.rs

```diff
@@ -62,6 +62,9 @@ pub fn execute_block<'alloc>(
     let mut data_cache = BlockDataCache::new(data_view);
     let mut result = vec![];
 
+    // TODO: MIRAI does not recognize that `result` is initially empty
+    assume!(result.is_empty());
```

Comment:
We need a foreign summary for the call that vec! expands to.

Reply:
In progress.